src/example_data.jl

```diff
@@ -0,0 +1,240 @@
+A synthetic multi dimensional (10 dimensions) logistic regression dataset with
+latent weights ("w") and intercept, passed into a Bernoulli random variable.
+Five hundred data points, fit with PyMC3.
+
+True weights and intercept are included as deterministic variables.""",
+    ),
+    "glycan_torsion_angles" => RemoteFileMetadata(;
+        name="glycan_torsion_angles",
+        filename="glycan_torsion_angles.nc",
+        url="http://figshare.com/ndownloader/files/22882652",
+        checksum="4622621fe7a1d3075c18c4c34af8cc57c59eabbb3501b20c6e2d9c6c4737034c",
+        description="""
+Torsion angles phi and psi are critical for determining the three dimensional
+structure of bio-molecules. Combinations of phi and psi torsion angles that
+produce clashes between atoms in the bio-molecule result in high energy, unlikely structures.
+
+This model uses a Von Mises distribution to propose torsion angles for the
+structure of a glycan molecule (pdb id: 2LIQ), and a Potential to estimate
+the proposed structure's energy. Said Potential is bound by Boltzman's law.""",
+    ),
+)
+
+"""
+    load_example_data(name; kwargs...) -> InferenceData
+    load_example_data() -> Dict{String,AbstractFileMetadata}
+
+Load a local or remote pre-made dataset.
+
+`kwargs` are forwarded to [`from_netcdf`](@ref).
+
+Pass no parameters to get a `Dict` listing all available datasets.
+
+Data files are handled by DataDeps.jl.
+A file is downloaded only when it is requested and then cached for future use.
+
+# Examples
+
+```jldoctest
+julia> keys(load_example_data())
+KeySet for a Dict{String, ArviZ.AbstractFileMetadata} with 9 entries. Keys:
+  "centered_eight"
+  "radon"
+  "glycan_torsion_angles"
+  "rugby"
+  "non_centered_eight"
+  "regression10d"
+  "classification1d"
+  "classification10d"
+  "regression1d"
+
+julia> load_example_data("centered_eight")
+InferenceData with groups:
+  > posterior
+  > posterior_predictive
+  > sample_stats
+  > prior
+  > observed_data
+```
+"""
+function load_example_data end
+function load_example_data(name::AbstractString; kwargs...)
+    metadata = get(EXAMPLE_DATA, name, nothing)
+    metadata === nothing && throw(
+        ArgumentError("Dataset $name not found. Valid datasets are: $(keys(EXAMPLE_DATA))"),
+    )
+    filename = joinpath(metadata.name, metadata.filename)
+    # none of our downloads require license agreements, so we always agree to the download
+    path = withenv("DATADEPS_ALWAYS_ACCEPT" => true) do
+        return DataDeps.resolve(filename, @__FILE__)
+    end
+    return from_netcdf(path; kwargs...)
+end
+load_example_data() = EXAMPLE_DATA
+
+function _register_data(md::LocalFileMetadata)
+    return DataDeps.register(DataDeps.ManualDataDep(md.name, md.description))
+end
+
+function _register_data(md::RemoteFileMetadata)
+    return DataDeps.register(DataDeps.DataDep(md.name, md.description, md.url, md.checksum))
+end
+
+_init_data_deps() = map(_register_data, values(EXAMPLE_DATA))
```

Comment:
@oxinabox, registering the 9 datasets in `EXAMPLE_DATA` seems to increase package load time by 300ms. That seems high. Am I using DataDeps incorrectly somehow?